src/settings/index.ts

```diff
@@ -60,81 +73,186 @@ export function useSettingsContext(): [Settings, SettingsSetters, Persistor] {
     return useContext(SettingsContext)
 }
 
+const getDocumentId = (): string | undefined => {
+    const id = window.location.pathname.match(ID_REGEX)
+
+    if (id) {
+        return id[1]
+    }
+}
+
 export function useSettings(): [Settings, SettingsSetters, Persistor] {
-    const [settings, setSettings] = useState(restore())
+    const [settings, setSettings] = useState<Settings>()
+
+    const documentId = getDocumentId()
+
+    const firebaseInitialized = useIsFirebaseInitialized()
+
+    const location = useLocation()
+
+    useEffect(() => {
+        if (location.pathname == '/' || !firebaseInitialized) return
+
+        async function loadSettings(): Promise<void> {
+            setSettings(await restore(documentId))
+        }
+
+        loadSettings()
+    }, [firebaseInitialized, location, documentId])
 
     const persistSettings = useCallback(() => {
-        persist(settings)
+        persist(settings, getDocumentId())
     }, [settings])
 
+    const updateFirebaseSetting = (
+        docId: string,
+        fieldId: string,
+        fieldValue:
+            | string
+            | number
+            | Array<string>
+            | firebase.firestore.GeoPoint
+            | { [key: string]: string[] },
+    ): Promise<void> => {
+        return updateSettingField(docId, fieldId, fieldValue)
+    }
+
     const set = useCallback(
         <T>(key: string, value: T, options?: SetOptions): void => {
             const newSettings = { ...settings, [key]: value }
             setSettings(newSettings)
             if (options && options.persist) {
-                persist(newSettings)
+                persist(newSettings, getDocumentId())
             }
         },
         [settings],
     )
 
     const setHiddenStations = useCallback(
         (newHiddenStations: Array<string>, options?: SetOptions): void => {
-            set('hiddenStations', newHiddenStations, options)
+            if (!documentId) {
+                set('hiddenStations', newHiddenStations, options)
+                return
+            }
+
+            updateFirebaseSetting(
```

Comment:
Det er egentlig `persist` som bør ha ansvar for persistering. Me har `UrlStorage.ts` som persist blir henta frå. Det er nok kanskje meste clean å lage ein `FirestoreStorage.ts` i tillegg, og importere begge:

```
 import { persist, restore } from './FirestoreStorage' 
 import { persist as persistInUrl, restore as restoreFromUrl } from './UrlStorage' 
```

... og så må man switche mellom "storage" i f.eks. `set`-funksjonen på linje 124.

```
if (options && options.persist) {
    if (getDocumentId()) {
        persist(newSettings)
    } else {
        persistToUrl(newSettings)
    }
}

Reply:
Det er nok en god løsning, ja! Ville unngå å bruke `persist` siden `URLStorage.ts` var såpass koblet til det, men det kan være greit å skille ut. Men om man gjør 'persist' inni `set`-funksjonen, vil ikke settings lagres kun når man trykker på lagre-knappen? Etter hva jeg forsto, var hensikten at settings lagres fortløpende.

Reply:
`set` blir kalt ved alle oppdateringar i admin, sidan alle funksjonane (f.eks. `setHiddenStations`) brukar denne. Til i dag har det vore mulig å sende ved `options`, som har eitt felt `persist: boolean`. `persist`-kallet skjer kun dersom denne er `true`.

Auto-save bør kanskje vente til lagre-knappen på Admin-sida er fjerna?

Reply:
Vi fikser autosave i en annen PR! I nyeste commit har vi nå revertert dette og splittet opp persisting mellom legacy og Firestore. Nå er dette altså i større grad splittet.